### PR TITLE
Change Feed Processor: Adds support for Resource Tokens

### PR DESCRIPTION
**Pull request by ealsur**
## Description

Currently the Change Feed Processor uses a `Read Database` call to obtain the Database Resource Id. Resource Tokens cannot perform this type of operation due to the nature of their scoped permissions, resulting in:

```
Forbidden, message: Insufficient permissions provided in the authorization header for the corresponding request. Please retry with another authorization header.
```

This PR modifies the code in a way that this call is no longer needed, the information can be inferred from the Container Resource Id.

This enables Resource Tokens and also reduces the network calls required to initialize the Processor.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Closing issues

Closes https://github.com/Azure/azure-cosmos-dotnet-v3/issues/3561

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/Utils/CosmosContainerExtensions.cs [hunk 1]
```diff
@@ -121,7 +121,11 @@ internal static class CosmosContainerExtensions
                 forceRefresh: false,
                 NoOpTrace.Singleton,
                 cancellationToken: cancellationToken);
-            string databaseRid = await ((DatabaseInternal)((ContainerInternal)monitoredContainer).Database).GetRIDAsync(cancellationToken);
```
[the remark this answers is not in the document]
Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/Utils/CosmosContainerExtensions.cs:124]: Non-blocking

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/Utils/CosmosContainerExtensions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/Utils/CosmosContainerExtensions.cs:124]: Done

**Review by kundadebdatta — APPROVED** [review]
LGTM.

**Pull request merged**